### PR TITLE
fix(ci): Fix network parameter in continous-delivery.yml, and add network labels to GCP jobs

### PR DESCRIPTION
**Pull request by teor2345**
## Motivation

We want to add the Zcash network as a GCP label. We need to lowercase the network to do that.

This also fixes a workflow input handling bug in continous-delivery.yml, which made it impossible to correctly implement a network label in PR #5693.

## Solution

- fix network parameter in continous-delivery.yml
- standardise network usage in zcashd-manual-deploy
- add Network as a label
- use lowercase network labels 

This PR also fixes some shellcheck errors using search and replace, including scripts in the unchanged parts of these workflows.

## Review

This is a routine change.

### Reviewer Checklist

  - [x] Will the PR name make sense to users?
    - [ ] Does it need extra CHANGELOG info? (new features, breaking changes, large changes)
  - [x] Are the PR labels correct?
  - [x] Does the code do what the ticket and PR says?
  - [x] How do you know it works? Does it have tests?

### CODE REVIEW

**Comment by teor2345** [comment]
@gustavovalverde I'm not sure how to test continous-delivery.yml, are there instructions?

zcashd-manual-deploy.yml isn't so important to test, because it is not used right now.

File: .github/workflows/continous-delivery.yml [hunk 1]
```diff
@@ -104,6 +104,17 @@ jobs:
...
 
+      # Makes the Zcash network name lowercase.
+      #
+      # Labels in GCP are required to be in lowercase, but the blockchain network
+      # uses sentence case, so we need to downcase ${{ inputs.network || 'Mainnet' }}.
+      #
+      # Passes the lowercase network to subsequent steps using $NETWORK env variable.
+      - name: Downcase network name for labels
+        run: |
+          NETWORK_CAPS="${{ inputs.network || 'Mainnet' }}"
```
Comment (arya2) [inline .github/workflows/continous-delivery.yml:115]: I'm not sure if the second `|| 'Mainnet'` is needed

Update: I mis-read, it is needed.

**Review by arya2 — COMMENTED** [review]
> I'm not sure how to test continous-delivery.yml

The workflow dispatch is available [here](https://github.com/ZcashFoundation/zebra/actions/workflows/continous-delivery.yml) if that helps.

**Comment by teor2345** [comment]
> > I'm not sure how to test continous-delivery.yml
> 
> The workflow dispatch is available [here](https://github.com/ZcashFoundation/zebra/actions/workflows/continous-delivery.yml) if that helps.

Thanks!

I started a manual run here:
https://github.com/ZcashFoundation/zebra/actions/runs/3536561823

But we also need to test the release trigger, and check the labels, and maybe the logs for the instances?

I'm not sure how to do any of that.

File: .github/workflows/zcashd-manual-deploy.yml [hunk 2]
```diff
@@ -37,6 +37,17 @@ jobs:
         with:
           short-length: 7
 
+      # Performs formatting on the Zcash network name.
+      #
+      # Labels in GCP are required to be in lowercase, but the blockchain network
+      # uses sentence case, so we need to downcase ${{ inputs.network }}.
+      #
+      # Passes ${{ inputs.network }} to subsequent steps using $NETWORK env variable.
+      - name: Format network name for labels
```
Comment (arya2) [inline .github/workflows/zcashd-manual-deploy.yml]: Nitpick:

```suggestion
      # Makes the Zcash network name lowercase.
      #
      # Labels in GCP are required to be in lowercase, but the blockchain network
      # uses sentence case, so we need to downcase ${{ inputs.network }}.
      #
      # Passes ${{ inputs.network }} to subsequent steps using $NETWORK env variable.
      - name: Downcase network name for labels
```

Comment (teor2345) [reply .github/workflows/zcashd-manual-deploy.yml]: Thanks, I think I accidentally overwrote that change with a force-push

**Comment by teor2345** [comment]
Failed due to #5384 again.

**Comment by teor2345** [comment]
@gustavovalverde I get this warning when running the workflow, but I don't think I made any changes to that part of the workflow:
> Deploy single instance
> The "create_credentials_file" option is true, but the current GitHub workspace is empty. Did you forget to use "actions/checkout" before this step? If you do not intend to share authentication with future steps in this job, set "create_credentials_file" to false.

https://github.com/ZcashFoundation/zebra/actions/runs/3536561823

**Comment by arya2** [comment]
> @gustavovalverde I get this warning when running the workflow, but I don't think I made any changes to that part of the workflow:

Thise seems to be normal, it says that for all runs going back at least ~6 months:

https://github.com/ZcashFoundation/zebra/actions/runs/2411825703

File: .github/workflows/zcashd-manual-deploy.yml (hunk 2 above)
Comment (arya2) [reply .github/workflows/zcashd-manual-deploy.yml]: > Thanks, I think I accidentally overwrote that change with a force-push

I think it happened again; I'll re-commit the suggestion.

**Review by arya2 — APPROVED** [review]
Looks good to me!

The "current GitHub workspace is empty" warning isn't new so I'm not worried about it.

Versioning and deploy-nodes were skipped in the manual run, but:
- versioning wasn't changed (except the comment)
- changes to deploy-nodes and zcashd-manual-deploy were tested in CI jobs

**Pull request merged**